File: pset2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

double coleman_index(string text); // Returns the Coleman-Liau Index of string text
int ispxq(char c);                 // Returns non-zero if char c is equal '.', '!' or '?'

int main(void)
{
    string text = get_string("Text: ");
    int grade = round(coleman_index(text));

    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

double coleman_index(string text)
{
    float letters = 0;
    float words = 0;
    float sentences = 0;

    // Loop to every char in the text
    for (int i = 0, length = strlen(text); i < length; i++)
    {
        char c = text[i];
        // Increment number of letters on every char a-z
        if (isalpha(c))
        {
            letters++;
        }
        // Increment number of words on every white space char, *//+1 in the end of the loop
        else if (isspace(c))
        {
            words++;
        }
        // Increment number of sentences on every period, exclamation point, or question mark
        else if (ispxq(c) && isalpha(text[i - 1]))
        {
            sentences++;
        }
    }
    words++; //*

    float L = (letters / words) * 100.0;
    float S = (sentences / words) * 100.0;
    double index = (0.0588 * L) - (0.296 * S) - 15.8;

    return index;
}

int ispxq(char c)
{
    char pxq[3] = {'.', '!', '?'};
    for (int i = 0; i < 3; i++)
    {
        if (c == pxq[i])
        {
            return 1;
        }
    }
    return 0;
}

File: pset6/sentimental-readability/readability.py
def main():
    text = input("Text: ")
    grade = round(coleman(text))

    if grade:
        if (grade < 1):
            print("Before Grade 1")
        elif (grade > 16):
            print("Grade 16+")
        else:
            print(f"Grade {grade}")
    else:
        print("Fail to check text.")


# Returns the Coleman-Liau Index of string text
def coleman(text):
    # Check if argument only contains digits
    if text.isdecimal():
        return False

    letters = 0
    words = 1  # Need to be initialized with 1
    sentences = 0
    ponct = ['.', '!', '?']

    # Loop to every char in the text
    for i in range(len(text)):
        c = text[i].lower()
        # Increment number of letters on every char a-z
        if c.isalpha():
            letters += 1
        # Increment number of words on every white space char
        elif c.isspace():
            words += 1
        # Increment number of sentences on every period, exclamation point, or question mark
        elif c in ponct:
            if i > 0 and text[i - 1].isalpha():
                sentences += 1

    L = (letters / words) * 100.0
    S = (sentences / words) * 100.0
    index = (0.0588 * L) - (0.296 * S) - 15.8

    print(index / 1)
    return index / 1


if __name__ == "__main__":
    main()
